middleware/auth.js

```diff
@@ -12,44 +12,79 @@ const jwtUtils = require("@utils/jwt-utils")
 // Instantiate router object
 const auth = express.Router()
 
+const { E2E_TEST_REPO, E2E_TEST_SECRET, E2E_TEST_GH_TOKEN } = process.env
+const E2E_TEST_USER = "e2e-test"
+const ALLOWED_URLS = ["/v1/sites", "/v1/auth/whoami"]
+
 function noVerify(req, res, next) {
   next("router")
 }
 
+function verifyE2E(req) {
+  const { isomercmsE2E } = req.cookies
+  const urlTokens = req.url.split("/") // urls take the form "/v1/sites/<repo>/<path>""
+  const repo = urlTokens[3]
+  let isValidE2E
+
+  if (isomercmsE2E) {
+    if (isomercmsE2E !== E2E_TEST_SECRET) throw new AuthError(`Bad credentials`)
+    if (!(repo === E2E_TEST_REPO || ALLOWED_URLS.includes(req.url)))
+      throw new AuthError(`E2E tests can only access the ${E2E_TEST_REPO} repo`)
```

Comment:
We could throw a different error if the route is outside of the allowed urls, since this error doesn't describe it very well

Reply:
Clarified this section with [b55c983](https://github.com/isomerpages/isomercms-backend/pull/270/commits/b55c983e06ea2c8fbad0645f6f0a2e372d302ebc)